### PR TITLE
Update glossary.yml

### PR DESCRIPTION
**Pull request by Kanga-Who**
Remove references to Hass.io and clarify descriptions not changed since rebranding.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_data/glossary.yml [hunk 1]
```diff
@@ -32,12 +32,12 @@
...
 - topic: hass
   description: "HASS or [hass](/docs/tools/hass/) is often used as an abbreviation for Home Assistant. It is also the command-line tool."
-- topic: Hass.io
-  description: "[Hass.io](/hassio/) is a full UI managed home automation ecosystem that runs Home Assistant, the Hass.io Supervisor and add-ons. It comes pre-installed on HassOS, but can be installed on any Linux system. It leverages Docker, which is managed by the Hass.io Supervisor."
-- topic: Hass.io Supervisor
-  description: "The Hass.io Supervisor is a program that manages a Hass.io installation, taking care of installing and updating Home Assistant, add-ons, itself and, if used, updating the HassOS operating system."
-- topic: HassOS
-  description: "HassOS, the Home Assistant Operating System, is an embedded, minimalistic, operating system designed to run the Hass.io ecosystem on single board computers (like the Raspberry Pi) or Virtual Machines. The Hass.io Supervisor can keep it up to date, removing the need for you to manage an operating system."
+- topic: Home Assistant Supervised (Deprecated Hass.io)
```
Comment (frenck) [inline source/_data/glossary.yml]:
```suggestion
- topic: Home Assistant Supervised (Previously Hass.io)
```

File: source/_data/glossary.yml [hunk 2]
```diff
@@ -32,12 +32,12 @@
...
+  description: "[Home Assistant](/hassio/) is a full UI managed home automation ecosystem that runs Home Assistant, the Home Assistant Supervisor and add-ons. It comes pre-installed on Home Assistant OS, but can be installed on any Linux system. It leverages Docker, which is managed by the Home Assistant Supervisor."
+- topic: Home Assistant Supervisor
+  description: "The Home Assistant Supervisor is a program that manages a Home Assistant installation, taking care of installing and updating Home Assistant, add-ons, itself and, if used, updating the HassOS operating system."
+- topic: Home Assistant OS
```
Comment (frenck) [inline source/_data/glossary.yml]: Let's not abbreviate, applies to other content in this PR as well.

```suggestion
- topic: Home Assistant Operating System
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_data/glossary.yml (hunk 2 above)
Comment (Kanga-Who) [reply source/_data/glossary.yml]: Agreed

**Review by frenck — APPROVED** [review]
Thanks, @Kanga-Who 👍

**Pull request merged**